swap/src/cli/watcher.rs

```diff
@@ -45,6 +48,25 @@ impl Watcher {
         loop {
             tokio::time::sleep(Duration::from_secs(Watcher::CHECK_INTERVAL)).await;
 
+            // Fetch the current Bitcoin balance
+            match self.wallet.balance().await {
+                Ok(new_balance) => {
+                    // Check if the balance has changed
+                    if new_balance != self.balance {
+                        // Emit the balance change event
+                        if let Some(tauri) = &self.tauri {
+                            tauri.emit_balance_change_event(new_balance);
+                        }
+                        // Update the stored balance
+                        self.balance = new_balance;
+                    }
+                }
+                // Log any errors
+                Err(e) => {
+                    tracing::error!("Failed to fetch Bitcoin balance: {}", e);
```

Comment:
nitpick: add the error as an attribute to the log message. It allows us to parse the error